scripts/combine/setupCombine.py

```diff
@@ -313,72 +281,115 @@ def assertSample(name, startsWith=["W", "Z"], excludeMatch=[]):
+            scale=1 if args.priorNormXsec < 0 else args.priorNormXsec, # histogram represents an (args.priorNormXsec*100)% prior
+            systAxes=poi_axes, 
+            mirror=True,
             passToFakes=passSystToFakes,
         )
-        if args.fitMassDiff == "charge":
-            cardTool.addSystematic(**mass_diff_args,
-                                # # on gen level based on the sample, only possible for mW
-                                # actionMap={m.name: (lambda h, swap=swap_bins: swap(h, "massShift", f"massShift{label}50MeVUp", f"massShift{label}50MeVDown")) 
-                                #     for g in cardTool.procGroups[signal_samples_forMass[0]] for m in cardTool.datagroups.groups[g].members if "minus" in m.name},
-                                # on reco level based on reco charge
-                                actionMap={m.name: (lambda h: 
-                                        hh.swap_histogram_bins(h, "massShift", f"massShift{label}50MeVUp", f"massShift{label}50MeVDown", "charge", 0)) 
-                                    for g in cardTool.procGroups[signal_samples_forMass[0]] for m in cardTool.datagroups.groups[g].members},
-            )
-        elif args.fitMassDiff == "eta-sign":
-            cardTool.addSystematic(**mass_diff_args, 
-                                actionMap={m.name: (lambda h: 
-                                        hh.swap_histogram_bins(h, "massShift", f"massShift{label}50MeVUp", f"massShift{label}50MeVDown", "eta", hist.tag.Slicer()[0:complex(0,0):]))
-                                    for g in cardTool.procGroups[signal_samples_forMass[0]] for m in cardTool.datagroups.groups[g].members},
-            )
-        elif args.fitMassDiff == "eta-range":
-            cardTool.addSystematic(**mass_diff_args, 
-                                actionMap={m.name: (lambda h: 
-                                        hh.swap_histogram_bins(h, "massShift", f"massShift{label}50MeVUp", f"massShift{label}50MeVDown", "eta", hist.tag.Slicer()[complex(0,-0.9):complex(0,0.9):]))
-                                    for g in cardTool.procGroups[signal_samples_forMass[0]] for m in cardTool.datagroups.groups[g].members},
+        if args.theoryAgnostic:
+            cardTool.addSystematic("yieldsUnfolding", 
+                **noi_args,
+                processes=["signal_samples_noOutAcc"], # currently not on out-of-acceptance signal template (to implement)
+                baseName=f"norm{label}CHANNEL_",
+                noConstraint=True if args.priorNormXsec < 0 else False,
+                #customizeNuisanceAttributes={".*AngCoeff4" : {"scale" : 1, "shapeType": "shapeNoConstraint"}},
+                labelsByAxis=["PtVBin", "YVBin", "AngCoeff"],
+                actionMap={
+                        m.name: (lambda h, scale: hh.addHists(
+                            h[{ax: hist.tag.Slicer()[::hist.sum] for ax in poi_axes}], 
+                            hh.scaleHist(h, scale)))
```

Comment:
On the first point I totally agree, I had it like this for the unfolding but forgot to adapt the same logic for the theory agnostic. I now modify these lines. 

On the second point, I see the current implementation as a simpler and less disruptive solution. Is there a problem with adding the overflow/underflow bins and the other helicity components to the histogram? You could of course add up the A5/6/7 beforehand so that you only have one additional component. But with the POI as NOI approach we get rid of having all the systematic and POI axes at the same time, so I would expect that the memory of the histogram is not an issue anymore. 

Yes, I would be happy if you can try it and see if you can live with it :) I push the changes addressing your comments now.